patches/chrome-browser-net-net_error_tab_helper.cc.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/chrome/browser/net/net_error_tab_helper.cc b/chrome/browser/net/net_error_tab_helper.cc
+index 0d094df161accd5740306543fc938ab6e58da296..eede25d41103e85be7c9067ecd09667a6c0e97b3 100644
+--- a/chrome/browser/net/net_error_tab_helper.cc
++++ b/chrome/browser/net/net_error_tab_helper.cc
+@@ -89,7 +89,8 @@ void NetErrorTabHelper::RenderFrameCreated(
+ 
+   chrome::mojom::NetworkDiagnosticsClientAssociatedPtr client;
+   render_frame_host->GetRemoteAssociatedInterfaces()->GetInterface(&client);
+-  client->SetCanShowNetworkDiagnosticsDialog(CanShowNetworkDiagnosticsDialog());
++  client->SetCanShowNetworkDiagnosticsDialog(
++      CanShowNetworkDiagnosticsDialog(web_contents()));
+ }
+ 
+ void NetErrorTabHelper::DidStartNavigation(
+@@ -259,6 +260,9 @@ void NetErrorTabHelper::InitializePref(WebContents* contents) {
+ }
+ 
+ bool NetErrorTabHelper::ProbesAllowed() const {
++#if defined(BRAVE_CHROMIUM_BUILD)
```

Comment:
this looks like it could be eliminated by using `testing_state_`?

Reply:
I assume this is not part of the upstream patch since it has `#if defined(BRAVE_CHROMIUM_BUILD)`?

Reply:
@bridiver yeah, this is the only block that is not part of upstream (it's in a separate commit)

Reply:
https://github.com/brave/brave-core/pull/1605/commits/34f4db5ed295ff0069b34caa627e98f882ab0643

Reply:
@bridiver it doesn't just depend on `testing_state_`: https://cs.chromium.org/chromium/src/chrome/browser/net/net_error_tab_helper.cc?q=ProbesAllowed&sq=package:chromium&g=0&l=267

but i guess setting `TESTING_FORCE_DISABLED` would fix the issue? is this possible to do without patching?

Reply:
it seems `set_state_for_testing` is meant to be used for the Chromium tests; while it would work here, i think it makes the change less clear (since we are not doing this for tests)